src/main/java/com/connexta/search/query/CommonQl.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.search.query;
+
+import com.connexta.search.common.configs.SolrConfiguration;
+import com.connexta.search.query.exceptions.IllegalQueryException;
+import com.connexta.search.query.exceptions.MalformedQueryException;
+import com.google.common.annotations.VisibleForTesting;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.validation.constraints.NotNull;
+import lombok.RequiredArgsConstructor;
+import org.geotools.filter.Filters;
+import org.geotools.filter.text.cql2.CQL;
+import org.geotools.filter.text.cql2.CQLException;
+import org.opengis.filter.Filter;
+import org.opengis.filter.expression.PropertyName;
+
+/**
+ * Class to encapsulate validating a CommonQL query string and building an OCG Filter object from
+ * the string. Create a new instance of this class by passing the CommonQL string into this the
+ * constructor. Please note this class is not thread safe. Each thread should create its own
+ * instance. Also, this class is not meant to be reused with different query strings and that is why
+ * there is not setter for the query string field.
+ */
+@RequiredArgsConstructor
+public class CommonQl {
+
+  @NotNull private final String queryString;
+  private Filter filter;
+
+  /**
+   * Attempt to validate the query string. Throw an exception is not valid. Check for supported
+   * attributes and throw an exception if an unsupported attribute is detected.
+   *
+   * @return this for fluent interface
+   * @throws IllegalQueryException
+   * @throws MalformedQueryException
```

Comment:
✏️ Declare these throws in the method signature. Do not include `@throws` in the javadoc if there is no description.

Reply:
Declaring them in the method signature results in SonarLint marking it as a minor code smell since it extends a RuntimeException. I'll add a description to each `throws`.